app/models/services/crossref/resource.rb

```diff
@@ -0,0 +1,52 @@
+class Crossref
+  class Resource
+    attr_accessor :id, :response
+
+    def initialize id
+      self.id = id.to_s
+      endpoint = URI.parse("http://api.crossref.org/works/#{self.id}")
+      self.response = Net::HTTP.get_response endpoint
+    end
+
+    def paper_attributes
+      if response.code_type.ancestors.include? Net::HTTPSuccess
+        data = JSON.parse response.body
+        @paper_attributes ||= map_attributes(mapper, data)
+      else
+        @paper_attributes = nil
+      end
+    end
+
+    def map_attributes mapper, data
+      return nil unless data
+      mapper.inject({}) do |memo, _|
+        attribute, mapping = _[0], _[1]
+        memo[attribute] = mapping.call(data)
+        memo
+      end
+    end
+
+    def mapper
+      {
+        title:              lambda {|data| data.dig 'message', 'title', 0 },
+        publication:        lambda {|data| data.dig 'message', 'short-container-title', 0 },
+        doi:                lambda {|data| self.id },
+        pubmed_id:          lambda {|data| Pubmed.get_uid_from_doi(id) },
```

Comment:
same comment I had yesterday here about checking if a pubmed ID already exists in our database but sounds we decided that will take place in the API imports meshing somewhere.